script/scriptManager.sh

```diff
@@ -0,0 +1,315 @@
+        read -p "enter chain id: " $SELECTED_CHAIN
+        arrayIndex=$(searchArray $SELECTED_CHAIN $CHAIN_ID_ARRAY)
+        L2_RPC_URL="${L2_ENDPOINTS_ARRAY[${arrayIndex}]}"
+        if [[ -z $L2_RPC_URL ]]; then
+            echo "Chain RPC not ound for $SELECTED_CHAIN"
+            exit 1
+        fi
+    fi
+    #exit if no url is found
+    if [[ -z $L2_RPC_URL ]]; then
+        exit 1
+    fi
+    echo $L2_RPC_URL
+
+}
+
+function checkInput() {
+    if [[ -z $2 || $2 == -* ]]; then
+        error_exit "Missing value for parameter $1"
+    fi
+}
+
+FILE_PATH=$1
+# if the first character of the path is a / trim it off
+if [[ "${FILE_PATH:0:1}" == "/" ]]; then
+    echo "trimming: $FILE_PATH"
+    FILE_PATH="${FILE_PATH:1}"
+fi
+echo "$FILE_PATH"
+# check that there is an arg
+if [[ -z $FILE_PATH ]]; then
+    display_help
+    exit 1
+    #if arg is a filepath and is a file set input json and shift down 1 arg
+elif [[ -f $FILE_PATH ]]; then
+    INPUT_JSON=$FILE_PATH
+    shift
+fi
+
+# gather flags used if you want to override the .env vars
+while [[ $# -gt 0 ]]; do
+    case $1 in
+    --l1-rpc-url | -l1)
+        checkInput $1 $2
+        L1_RPC_URL=$2
+        shift 2
+        ;;
+        # pass in an array of rpc's
+    --l2-rpc-urls | -l2)
+        checkInput $1 $2
+        L2_ENDPOINTS_ARRAY=$2
+        shift 2
+        ;;
+    --etherscan-api-key | -api)
+        checkInput $1 $2
+        ETHERSCAN_API_KEY=$2
+        shift 2
+        ;;
+    --deployer-account-name | -a)
+        checkInput $1 $2
+        DEPLOYER_ACCOUNT_NAME=$2
+        shift 2
+        ;;
+    --sender-account-address | -s)
+        checkInput $1 $2
+        DEPLOYER_ADDRESS=$2
+        shift 2
+        ;;
+    --help | -h)
+        display_help
+        exit 0
+        ;;
+    *)
+        echo "Error, unrecognized flag" >&2
+        display_help
+        exit 1
+        ;;
+    esac
+done
+
+L1_CHAIN_ID=$(jq -r ".l1ChainId" "$INPUT_JSON")
+ERC20_NAME=$(jq -r ".erc20Name" "$INPUT_JSON")
+L2_CHAIN_IDS_ARRAY=$(jq -r ".l2ChainIds" "$INPUT_JSON" | jq -r ".[]")
+L2_ENDPOINTS_ARRAY=$(< <(getRpcEndpoints $L2_CHAIN_IDS_ARRAY))
+echo "$L2_ENDPOINTS_ARRAY"
+
+# if rpc url is not set by flag set it here
+if [[ -z $L1_RPC_URL ]]; then
+    L1_RPC_URL=$(getEndpoint $L1_CHAIN_ID)
+fi
+
+# verify the l1 rpc has been set
+if [[ -z $L1_RPC_URL || $L1_RPC_URL == "Unrecognized Chain Id" ]]; then
+    echo "Valid rpc required"
+    exit 1
+fi
+
+# if the number of rpc's is less than the number of L2 chain ids
+if [[ "${#L2_CHAIN_IDS_ARRAY[@]}" != "${#L2_ENDPOINTS_ARRAY[@]}" ]]; then
+    echo "Invalid L2 RPCs"
+    exit 1
+fi
+
+# forge script script/DeployL1OFTAdapter.s.sol:DeployL1OFTAdapter \
+# --rpc-url ${rpc} --sig "run(string calldata)" ${path} \
+# --account ${deployerAccountName} --sender ${deployer} \
+# --broadcast --etherscan-api-key ${api} --verify
+
+CALLDATA=$(cast calldata "run(string)" "/$FILE_PATH")
+clear
+echo "What would you like to deploy?"
```

Comment:
Instead of asking the user for what they want to deploy we should rather deploy all the contracts in sequence. Since we can read the saved deployment file then we know if some contracts are already deployed or not, so we should be able to determine if a new chain is being added or if it already exists. And we determine the flow of the scripts based on that.